src/core/net/tcp6.cpp

```diff
@@ -846,6 +920,25 @@ void Tcp::ProcessTimers()
     }
 }
 
+void Tcp::HandleTasklet(Tasklet &aTasklet)
+{
+    OT_ASSERT(&aTasklet == &aTasklet.GetInstance().Get<Tcp>().mTasklet);
+    LogDebg("TCP tasklet invoked");
+    aTasklet.GetInstance().Get<Tcp>().ProcessCallbacks();
```

Comment:
I think we don't need the `GetInstance()` and can call use `Get<Tcp>()` directly on `aTasklet`.
```suggestion
    aTasklet.Get<Tcp>().ProcessCallbacks();
```

Reply:
I think you may need to add `#include "common/locator_getters.hpp"` for this in `tcp6.cpp` file.

Reply:
Fixed in the latest update (and yes, I had to include that header).